src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -89,24 +119,86 @@ public void remove(Person toRemove) {
         if (!internalList.remove(toRemove)) {
             throw new PersonNotFoundException();
         }
+        removeLessons(toRemove.getLessons());
+    }
+
+    /**
+     * Removes the equivalent Lesson from the Calendar.
+     * The lesson must exist in the calendar.
+     */
+    public void removeLesson(Person target, Person editedPerson, Lesson toRemove) {
+        requireAllNonNull(target, editedPerson, toRemove);
+
+        setPerson(target, editedPerson);
     }
 
     public void setPersons(UniquePersonList replacement) {
         requireNonNull(replacement);
         internalList.setAll(replacement.internalList);
+        setLessons(replacement);
     }
 
     /**
      * Replaces the contents of this list with {@code persons}.
      * {@code persons} must not contain duplicate persons.
+     *
+     * @param persons The Persons to be set.
      */
     public void setPersons(List<Person> persons) {
         requireAllNonNull(persons);
         if (!personsAreUnique(persons)) {
             throw new DuplicatePersonException();
         }
-
         internalList.setAll(persons);
+        setLessons(persons);
+    }
+
+    public void setLessons(UniquePersonList replacement) {
+        requireNonNull(replacement);
+        setLessons(replacement.internalList);
+    }
+
+    /**
+     * Replaces the entries of the calendar with {@code persons}.
+     * {@code persons} must not contain clashing lessons.
+     */
+    public void setLessons(List<Person> persons) {
+        requireAllNonNull(persons);
+        calendar.clear();
+        for (Person person : persons) {
+            addLessons(person.getLessons());
+        }
+    }
+
+    /**
+     * Adds all specified lessons into the calendar.
+     *
+     * @param lessons The lessons to be added.
+     */
+    public void addLessons(Set<Lesson> lessons) {
+        requireAllNonNull(lessons);
+        calendar.startBatchUpdates();
+        for (Lesson lesson : lessons) {
+            calendar.addEntry(lesson.asCalendarEntry());
+        }
+        calendar.stopBatchUpdates();
+    }
```

Comment:
these calendar related fields and methods feel a bit out of place in `UniquePersonList`, should we have like a `CalendarManager`. If we want to do more functions with calendar in future it'll be easier too. Can do this in next iteration also if not enough time.